### DISCUSSION TITLE
Loading a pickled `xarray` object

### DISCUSSION THREAD

**Discussion by tommylees112** [pydata/xarray#5642] [Q&A] [answered]
I am trying to open an old pickled xarray object. I get the following errors:

```python
~/neuralhydrology/scripts/read_nh_results.py in get_all_station_ds(res_fp)
    115
    116 def get_all_station_ds(res_fp: Path) -> xr.Dataset:
--> 117     res_dict = pickle.load(res_fp.open("rb"))
    118     ds = _load_dict_to_xarray(res_dict)
    119     return ds

AttributeError: Can't get attribute 'PandasIndexAdapter' on <module 'xarray.core.indexing' from '/DataDrive200/data/miniconda3/envs/ml/lib/python3.8/site-packages/xarray/core/indexing.py'>
```

Is there any way of determining which versions of xarray the file was saved with? Or else determining how to go about opening the file?

My package versions:
```
xr.__version__
'0.19.0'

pd.__version__
'1.3.1'

python=3.8.10
```

## Steps to reproduce:

The pickle file can be found [here](https://drive.google.com/file/d/1piiq9vmvL5OVw3ppjZWVNsSmAdBq66jw/view?usp=sharing).

```python
import pickle
from pathlib import Path

fp = Path("path/to/file/test_results.p")
pickle.load(fp.open("rb"))
```

**Comment by jhamman** [comment, marked as answer]
Unfortunately, Xarray does not provide backward compatibility for Pickled objects. In fact, the [Xarray documentation on the Pickle format says](http://xarray.pydata.org/en/stable/user-guide/io.html#pickle):  

>... pickling is not recommended for long-term storage.
>
>Restoring a pickle requires that the internal structure of the types for the pickled data remain unchanged. Because the internal design of xarray is still being refined, we make no guarantees (at this point) that objects pickled with this version of xarray will work in future versions.

----

I take it from your question that you don't have the environment that created the Pickle objects? I am not aware of any tools for extracting environment details from a Pickle.

**Reply by tommylees112** [reply]
Thanks so much ! No worries i thought this might be the case, i just wondered if anyone had a super smart solution :) 

thank you again

**Reply by Meelisha** [reply]
Hi! Were you able to find a solution for this?
